web/router.js

```diff
@@ -145,31 +159,29 @@ async function beforeEachCheckAuth(to) {
      */
     const metaAuth = routeMetaKey(to, 'auth', true);
     if (metaAuth === true) {
-      // this route requires an authenticated user
-      if (loggedIn) {
-        return false;
+      if (!loggedIn) {
+        store.dispatch('setToast', ROUTE_NOT_LOGGED_IN);
       }
-      store.dispatch('setToast', ROUTE_NOT_LOGGED_IN);
-      return { name: 'home', query: { path, login: true } };
+      return loggedIn;
     }
     if (metaAuth === false) {
       // this route requires an unauthenticated user
-      return loggedIn ? { name: 'home' } : false;
+      return !loggedIn;
     }
-    return false;
+    return true;
```

Comment:
now that we have the `onError` handler in `router`, we shouldn't need to catch the error here - having more unified handling of this is a Good Thing